### PR TITLE
Set up a "Hello world" project with two apps

### PR DESCRIPTION
**Pull request by tchasinga**
Implement a connection between a Ruby on Rails backend and the React frontend.
Understand the pros and cons of different approaches to connecting the Ruby on Rails backend with the React frontend.

### CODE REVIEW

**Review by Davidosky007 — COMMENTED** [review]
Hi @tchasinga 👋, 

While you made a great effort in this project, unfortunately, I cannot proceed to review your code.

## Invalid Code Review Request

You have submitted a project, that has a different frontend react [app link](https://github.com/meryemsanem/hello-react-front-end/tree/741c67326e6581e6efeecfb1ff204dc7a19d2c44) as the link took me to someone else project, i assume this might be a mistake, kindly add the right frontend app link to your readme 

Your Code Review Request will be marked as invalid in your Dashboard, so please submit a new one once you are ready 🙏

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

--------

_Invalid Code Review Request **does not count** into the code reviews limit._

File: db/seeds.rb [hunk 1]
```diff
@@ -0,0 +1,5 @@
+Greeting.create(message: "Welcome to my website!")
+Greeting.create(message: "Hey there!")
+Greeting.create(message: "Good day!")
+Greeting.create(message: "Greetings #1!")
+Greeting.create(message: "Greetings #2!")
```
Comment (AmaduKamara) [inline db/seeds.rb]: - [ ] When I ran `rails db:seed` command to generate random greetings, I got an `UnknownAttributeError` - unknown attribute 'message' for Greeting. Kindly work on the attribute name so the seed command runs fine and random greetings are generated for the front-end app to show them as expected 👍🏽

Screenshot 2023-11-03 at 6 22 38 PM

**Review by AmaduKamara — CHANGES_REQUESTED** [review]
# Status: Required Changes ♻️ ♻️ ♻️

Hi @tchasinga,

Good job so far!
There are some issues that you still need to work on to go to the next project but you are almost there!

## Highlights

✅ You set up Gitflow correctly 👍🏽
✅ The Linter checks passed successfully 👍🏽
✅ Add the correct links to the two repositories in the README as required 👍🏽

## Required Changes ♻️
_Check the comments under the review._

- [ ] Kindly work on the seed error showing so greetings are generated and shown on the front-end app 👍🏽

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you take them into account as they can make your code better._

Cheers, and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._
```

**Review by frank1738 — CHANGES_REQUESTED** [review]
Hi @tchasinga , 

Good job so far!
There are some issues that you still need to work on to go to the next project but you are almost there!



## Highlights

✅ The backend API endpoint is generating a random greeting
✅ Good work updating your the `seed data`


#### Keep up the good work! 😊🚀

## Required Changes ♻️
- [ ] 🌐 Just a quick heads-up! The URL you're using to fetch a random greeting in your Front End is set to the root of your backend API. To get that random greeting, you should update the URL to `http://localhost:3000/random_greeting`. Simply append `/random_greeting` to your URL, and you'll be all set .👍
- [ ]   💡 After you've fetched the greeting and updated the state, you can directly use `greetingsCont.greeting` in your greeting component to display the random greeting. The fetched greeting is already random, so there's no need to use `Math.random()` on the data.
- [ ]  👇 Here's an example of how your component could look based on your work:

```
function Greetings() {
  const dispatch = useDispatch();
  const greetingsCont = useSelector((state) => state.greetings.greetings);

  useEffect(() => {
    dispatch(fetchGreetings());
  }, [dispatch]);

  return (
    <div>
      <h2>{greetingsCont.greeting}</h2>
    </div>
  );
}

export default Greetings;


```

- [ ]  This way, you'll simplify your code and still display the random greetings you've fetched. Keep up the good work! 😊🚀

## Optional suggestions

- [ ] Instead of simply naming your branch "newCod," how about giving it a descriptive name that aligns with the tasks for this milestone? for example, we can use `generateRandomGreetingAPI`

- [ ] 📝 Remember to write your Git commit messages in imperative tense and capitalize them. For example, `Add feature to generate random greeting endpoint` is a great way to follow this convention. 👍

- [ ] 👀 I noticed the front-end repo doesn't quite follow Git flow practice. It looks like you compared the dev branch to the main branch. Remember, with Gitflow, feature branches should be compared to the dev/development branch. Let's stick to these best practices to make our work more appealing to potential recruiters! 🌟😊

_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._


Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Review by shella12 — APPROVED** [review]
Hi @tchasinga, 


## Status : APPROVED 🥳 🥳 

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉 



## Highlights

- Fixed URL ✔️ 
- Removed `Math.random` 👍 
- Implemented all requirements 👏 

## Optional suggestions

_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**